app/components/search.tsx

```diff
@@ -1,36 +1,69 @@
-import {useState, useEffect, useRef, MutableRefObject, FocusEvent} from 'react'
+import {useState, useEffect, useCallback, useRef, MutableRefObject, FocusEvent} from 'react'
 import debounce from 'lodash/debounce'
+import {Searcher, Question as QuestionType, SearchResult} from 'stampy-search'
 import {AddQuestion} from '~/routes/questions/add'
 import {Action, ActionType} from '~/routes/questions/actions'
 import {MagnifyingGlass, Edit} from '~/components/icons-generated'
-import {useSearch, Question as QuestionType, SearchResult} from '~/hooks/search'
 import AutoHeight from 'react-auto-height'
 import Dialog from '~/components/dialog'
 
 type Props = {
   onSiteAnswersRef: MutableRefObject<QuestionType[]>
+  initialQuery?: string
   openQuestionTitles: string[]
   onSelect: (pageid: string, title: string) => void
 }
 
 const empty: [] = []
 
-export default function Search({onSiteAnswersRef, openQuestionTitles, onSelect}: Props) {
-  const [showResults, setShowResults] = useState(false)
+export default function Search({
+  onSiteAnswersRef,
+  initialQuery,
+  openQuestionTitles,
+  onSelect,
+}: Props) {
+  const [showResults, setShowResults] = useState(initialQuery !== undefined)
   const [showMore, setShowMore] = useState(false)
   const searchInputRef = useRef('')
 
-  const {search, arePendingSearches, results} = useSearch(onSiteAnswersRef)
+  const [arePendingSearches, setPendingSearches] = useState(false)
+  const [results, setResults] = useState([] as SearchResult[])
+  const [searcher, setSearcher] = useState<Searcher>()
+
+  useEffect(() => {
+    setSearcher(
+      new Searcher({
+        getAllQuestions: () => onSiteAnswersRef.current,
```

Comment:
This looks like a very useful exercise to find out how reusable is current architecture, and I'd be happy for this code to live on a branch or fork for experimental deployments.

However, I am not able to confirm this is the right step for the master branch and production deployment, where we should aim for more maintainability and simplified architecture. Adding more features on top of previous hacks does not sound sustainable to me.

For 3rd party users of this feature, creating new class instances to spin of workers sensitive to CORS and special API calls while having to provide your own `getAllQuestions`, then dealing with optional-results success callbacks without error handling and quite complex semantics ... that doesn't sound very simple to use / modular / ...

If there is someone else who wants to take responsibility for long term maintenance of this repo with occational firefighting and debugging if the site stops working, I'd be happy to hand over the "architecture oversight".

If not, I will rather focus on figuring out how to include stampy search on 3rd party sites using some simpler API, without this new PoC feature 🤔 Ideally something drop-in replacable with API from https://nlp.stampy.ai/ (or whatever is / will be the current url / name for that feature)